### PR TITLE
Implement Commit and tests

### PR DESCRIPTION
**Pull request by QiWang19**

### CODE REVIEW

File: src/ContainerCommitModal.jsx [hunk 1]
```diff
@@ -67,9 +67,10 @@ class ContainerCommitModal extends React.Component {
         });
     }
 
-    // TODO
     handleCommit() {
-
+        const commitMsg = {...this.state};
```
Comment (KKoukiou) [inline src/ContainerCommitModal.jsx]: commitMsg is used only here. I think it makes sense to use this.state directly here.

File: src/Containers.jsx [hunk 2]
```diff
@@ -79,9 +79,43 @@ class Containers extends React.Component {
...
+        }
+        commitData.changes.push(...onbuildsArr);
 
+        utils.varlinkCall(utils.PODMAN, "io.podman.Commit", commitData)
+                .then(reply => {
+                    this.props.updateImagesAfterEvent();
+                    this.props.updateContainersAfterEvent();
+                })
+                .catch(ex => {
+                    console.error("Failed to do Commit call:", ex, JSON.stringify(ex));
```
Comment (KKoukiou) [inline src/Containers.jsx]: If Commit call failed the user should see error notification in the open dialog.
In other words the dialog wondow should not close IMO if:
* Commit varlinkCall failed
* Some of the options are invalid (we can perform some empty string checks there for example)

File: test/check-application [hunk 3]
```diff
@@ -93,9 +93,26 @@ class TestApplication(testlib.MachineCase):
...
+        b.set_input_text("#author", "tester")
+        b.set_input_text("#command", "sleep 6000")
+        b.click(".modal-dialog div .btn-ctr-commit")
+        b.wait_not_present(".modal-dialog div")
+        b.wait_present('#containers-images tr:contains("testimg:testtag")')
+        b.click('#containers-images tbody tr:contains("testimg:testtag") td.listing-ct-toggle')
+        # open the listing toggle of testimg and check the commit paramerters
+        b.wait_present('#containers-images tbody tr:contains("testimg:testtag"):has(dd:contains("localhost/testimg:testtag"))')
+        b.wait_present('#containers-images tbody tr:contains("testimg:testtag"):has(dd:contains("sleep 6000"))')
+        b.wait_present('#containers-images tbody tr:contains("testimg:testtag"):has(dd:contains("tester"))')
```
Comment (KKoukiou) [inline test/check-application]: There should be also test for failed image creation from commited container.
Notification in the UI should be checked in this case.

**Comment by QiWang19** [comment]
@KKoukiou PTAL

File: src/Containers.jsx [hunk 4]
```diff
@@ -79,9 +83,60 @@ class Containers extends React.Component {
...
+            onbuildsArr = utils.getCommitArr(commitMsg.onbuild, "ONBUILD");
+        }
+        commitData.changes.push(...onbuildsArr);
+
+        utils.varlinkCall(utils.PODMAN, "io.podman.Commit", commitData)
+                .then(reply => {
+                    this.props.updateImagesAfterEvent();
+                    this.props.updateContainersAfterEvent();
+                    this.setState((prevState) => ({
+                        setContainerCommitModal: !prevState.setContainerCommitModal
```
Comment (KKoukiou) [inline src/Containers.jsx]: Is there reason that you take here the prevState and not just set it to false?

**Comment by QiWang19** [comment]
:+1: I like it. Do you know how to set the default value of input field in `<FormControl/>`? I wanted to set command value there but failed.

**Pull request merged**